File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        nums = sorted(list(set(nums)))
        count = 1
        temp = 1
        print(nums)
        for i in range(len(nums) - 1):
            if nums[i] + 1 == nums[i+1]:
                temp += 1
                print("nums[i]: ", nums[i])
                print("temp:", temp)
                print("count: ", count)
            else:
                count = max(count, temp)
                temp = 1
        count = max(count, temp)
        return count

File: Data Structures & Algorithms/maximum-subarray/submission-4.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int globalMax = nums[0], curSum = nums[0];

        for (int i = 1; i < nums.size(); ++i) {
            curSum = max(nums[i], curSum + nums[i]);
            globalMax = max(globalMax, curSum);
        }
        return globalMax;
    }
};
